payjoin/src/send/mod.rs

```diff
@@ -550,16 +568,98 @@ mod test {
+            Some((Amount::from_sat(1000), None)),
+            true,
+        );
+        assert_eq!(fee_contribution.err(), Some(InternalBuildSenderError::NoOutputs));
+
+        // Psbt with identical receiver outputs
+        let mut psbt = PARSED_ORIGINAL_PSBT.clone();
+        psbt.outputs[1] = psbt.outputs[0].clone();
+        psbt.unsigned_tx.output[1].script_pubkey = psbt.unsigned_tx.output[0].script_pubkey.clone();
+
+        let fee_contribution = determine_fee_contribution(
+            &psbt,
+            &ScriptBuf::from_hex("a9141de849f069d274150e3afeae8d72eb5a6b09443087")?,
+            Some((Amount::from_sat(1000), None)),
+            true,
+        );
+        assert_eq!(fee_contribution.err(), Some(InternalBuildSenderError::MultiplePayeeOutputs));
+
+        // Psbt with only one output
+        let mut psbt = PARSED_ORIGINAL_PSBT.clone();
+        psbt.outputs.pop();
+        psbt.unsigned_tx.output.pop();
+
+        let fee_contribution = determine_fee_contribution(
+            &psbt,
+            Script::from_bytes(
+                &<Vec<u8> as FromHex>::from_hex("a9141de849f069d274150e3afeae8d72eb5a6b09443087")
+                    .unwrap(),
+            ),
+            Some((Amount::from_sat(1000), None)),
+            true,
+        );
+        assert_eq!(fee_contribution, Ok(None));
+
+        let fee_contribution = determine_fee_contribution(
+            &psbt,
+            Script::from_bytes(
+                &<Vec<u8> as FromHex>::from_hex("a9141de849f069d274150e3afeae8d72eb5a6b09443087")
+                    .unwrap(),
+            ),
+            Some((Amount::from_sat(1000), None)),
+            false,
+        );
+        assert_eq!(
+            fee_contribution.err(),
+            Some(InternalBuildSenderError::FeeOutputValueLowerThanFeeContribution)
+        );
+
+        let fee_contribution = determine_fee_contribution(
+            &psbt,
+            &payee_script,
+            Some((Amount::from_sat(1000), None)),
+            false,
+        );
+        assert_eq!(fee_contribution.err(), Some(InternalBuildSenderError::MissingPayeeOutput));
+
+        let fee_contribution = determine_fee_contribution(
+            &psbt,
+            &payee_script,
+            Some((Amount::from_sat(1000), None)),
+            true,
+        );
+        assert_eq!(fee_contribution.err(), Some(InternalBuildSenderError::MissingPayeeOutput));
+
+        // Psbt with three total outputs
+        let mut psbt = PARSED_ORIGINAL_PSBT.clone();
+        psbt.outputs.push(psbt.outputs[1].clone());
+        psbt.unsigned_tx.output.push(psbt.unsigned_tx.output[1].clone());
+
+        let fee_contribution = determine_fee_contribution(
+            &psbt,
+            &payee_script,
+            Some((Amount::from_sat(1000), None)),
+            true,
+        );
+        assert_eq!(fee_contribution.err(), Some(InternalBuildSenderError::AmbiguousChangeOutput));
```

Comment:
In the future I think this syntax is simpler:

```suggestion
        assert_eq!(fee_contribution, Err(InternalBuildSenderError::AmbiguousChangeOutput));
```